homeassistant/components/synology_dsm/camera.py

```diff
@@ -55,8 +57,11 @@ class SynoDSMCamera(SynologyDSMBaseEntity, Camera):
     """Representation a Synology camera."""
 
     def __init__(
-        self, api: SynoApi, coordinator: DataUpdateCoordinator, camera_id: int
-    ):
+        self,
```

Comment:
Add `coordinator: DataUpdateCoordinator[dict[str, dict[str, SynoCamera]]]` to redefine type for `self.coordinator`. Like described here: https://github.com/home-assistant/core/pull/49898#issuecomment-830621906